src/main/java/org/sonarlint/intellij/core/AnalysisRequirementNotifications.java

```diff
@@ -24,25 +24,96 @@
 import org.sonarlint.intellij.ui.SonarLintToolWindowFactory;
+import org.sonarsource.sonarlint.core.client.api.common.Language;
+import org.sonarsource.sonarlint.core.client.api.common.PluginDetails;
+import org.sonarsource.sonarlint.core.client.api.common.SkipReason;
+import org.sonarsource.sonarlint.core.client.api.common.analysis.AnalysisResults;
 
 public class AnalysisRequirementNotifications {
 
-  private static final NotificationGroup ANALYSIS_PROBLEM_GROUP = NotificationGroup.balloonGroup("SonarLint: Analysis Requirement");
+  private static final NotificationGroup ANALYZER_REQUIREMENT_GROUP = NotificationGroup.balloonGroup("SonarLint: Analyzer Requirement");
+
+  private static final Set<String> alreadyNotified = new HashSet<>();
 
   private AnalysisRequirementNotifications() {
     // NOP
   }
 
+  public static void notifyOnceForSkippedPlugins(AnalysisResults analysisResults, Collection<PluginDetails> allPlugins, Project project) {
+    Set<Language> attemptedLanguages = new HashSet<>(analysisResults.languagePerFile().values());
+    attemptedLanguages.forEach(l -> {
+      final Optional<PluginDetails> correspondingPlugin = allPlugins.stream().filter(p -> p.key().equals(l.getPluginKey())).findFirst();
+      correspondingPlugin.flatMap(PluginDetails::skipReason).ifPresent(skipReason -> {
+        if (skipReason instanceof SkipReason.UnsatisfiedRuntimeRequirement) {
+          final SkipReason.UnsatisfiedRuntimeRequirement runtimeRequirement = (SkipReason.UnsatisfiedRuntimeRequirement) skipReason;
+          final String title = "<b>SonarLint failed to analyze " + l.getLabel() + " code</b>";
+          if (runtimeRequirement.getRuntime() == SkipReason.UnsatisfiedRuntimeRequirement.RuntimeRequirement.JRE) {
+            String content = String.format("SonarLint requires Java runtime version %s or later to analyze %s code. Current version is %s.<br>See <a href=\"https://intellij-support.jetbrains.com/hc/en-us/articles/206544879-Selecting-the-JDK-version-the-IDE-will-run-under\">how to select the JDK version the IDE will run under</a>.",
+              runtimeRequirement.getMinVersion(), l.getLabel(), runtimeRequirement.getCurrentVersion());
+            createNotificationOnce(project, title, content, new NotificationListener.UrlOpeningListener(true));
+          } else if (runtimeRequirement.getRuntime() == SkipReason.UnsatisfiedRuntimeRequirement.RuntimeRequirement.NODEJS) {
+            StringBuilder content = new StringBuilder(String.format("SonarLint requires Node.js runtime version %s or later to analyze %s code.", runtimeRequirement.getMinVersion(), l.getLabel()));
+            if (runtimeRequirement.getCurrentVersion() != null) {
+              content.append(String.format(" Current version is %s.", runtimeRequirement.getCurrentVersion()));
+            }
+            content.append("<br>Please configure the Node.js path in the <a href=\"#edit-settings\">SonarLint settings</a>.");
```

Comment:
I guess the new settings will come in another PR?

Reply:
in fact I added it to the same PR, sorry if it makes the review more difficult